pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -695,6 +696,16 @@ func getServersConfig(cr *api.KieApp) ([]api.ServerTemplate, error) {
 			}
 
 			// Set replicas
+			// if the configuration use only one replica we increment
+			if len(cr.Spec.Objects.Servers) != 0 && cr.Spec.Objects.Servers[0].JbpmCluster && *serverReplicas == 1 {
```

Comment:
At this point of the code base you're already in the servers range and already know that len of Servers is greater than 1. are we sure that check is necessary? also, try to use serverSet whenever possible... e.g. `serverSet.JbpmCluster`.

Reply:
Added because some test fails